controllers/imageclusterinstall_controller.go

```diff
@@ -246,6 +266,49 @@ func (r *ImageClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.
 	return ctrl.Result{}, nil
 }
 
+func (r *ImageClusterInstallReconciler) validateSeedReconfigurationWithBMH(
+	ici *v1alpha1.ImageClusterInstall,
+	bmh *bmh_v1alpha1.BareMetalHost) error {
+
+	// no need to validate if inspect annotation is disabled
+	if bmh.ObjectMeta.Annotations != nil && bmh.ObjectMeta.Annotations[inspectAnnotation] == "disabled" {
+		r.Log.Infof("inspection is disabled for BareMetalHost %s/%s, skip hardware validation", bmh.Namespace, bmh.Name)
+		return nil
+	}
+
+	clusterInfoFilePath, err := r.clusterInfoFilePath(ici)
+	if err != nil {
+		return err
+	}
+	clusterInfo := r.getClusterInfoFromFile(clusterInfoFilePath)
+
+	if bmh.Status.HardwareDetails == nil {
+		return fmt.Errorf("hardware details not found for BareMetalHost %s/%s", bmh.Namespace, bmh.Name)
```

Comment:
I know I said we'll rerun reconcile on an error, but this isn't really an error, right?
We know we're waiting on some external condition so I think it would be better to log an info and requeue here.

Reply:
Actually i am not sure about it. Maybe lets log it but still set error? like Waiting for host hardware details?

Reply:
This should update the status to pending and add this error as the message + requeue

Reply:
We need a new condition, to what we have in ACI - https://github.com/openshift/assisted-service/blob/8a63485495f366dbdf45b33c31fd0994e0c0986a/api/hiveextension/v1beta1/agentclusterinstall_types.go#L33